src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -33,9 +32,15 @@
                 .onItem().call(() -> {
                     // If the endpoint properties are null, they won't be persisted.
                     if (endpoint.getProperties() != null) {
+                        /*
+                         * As weird as it seems, we need the following line because the Endpoint instance was
+                         * deserialized from JSON and that JSON did not contain any information about the
+                         * @OneToOne relation from EndpointProperties to Endpoint.
+                         */
+                        endpoint.getProperties().setEndpoint(endpoint);
```

Comment:
Is this the only method that receives the `Endpoint` ?  (I'm actually talking about the REST API that reaches up to this point)
or maybe we need to add this logic as part of the de-serialization process?

Reply:
That's the only one where we have to do that. `PUT /endpoints/{id}` also receives an `Endpoint` but in that case the relation between `EndpointProperties` and `Endpoint` already exists.

We could do something during the deserialization, but that would probably mean writing multiple custom Jackson deserializers (one for each type of property) and duplicating the code that establishes the missing relation in each deserializer. That would be a lot of code to get rid of that single line :)

Reply:
Then it's fine to have it here I think.